### PR TITLE
Fix(eos_designs): WAN Preview - Update configs to align to best practices

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

WAN Preview - Update configs to align to best practices

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

Implementing the following fixes for WAN routers:

- Use Dps1 interface IP as VTEP, BGP source, IKE ID and DPS endpoint.
- Configure TCP MSS auto detection under router path-selection instead of under Dps1. Use "auto" instead of default value of 1000.
- Ensure only Loopback0 is exported to EVPN.
  - Later we need to add routes learned form uplink bgp when applicable.
- Ensure only Loopback0 is redistributed to BGP for the default VRF.

## How to test

Changes reflected in existing molecule.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/application_traffic_recognition.py [hunk 1]
```diff
@@ -47,8 +47,8 @@ def _wan_cp_app_src_prefix(self) -> str:
     def _generate_control_plane_application_profile(self, app_dict: dict) -> None:
         """
         Generate an application profile using a single application matching:
-        * the device Pathfinders router_ids  as destination for non Pathfinders.
-        * the device Pathfinder router_id as source
+        * the device Pathfinders vtep_ips  as destination for non Pathfinders.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/application_traffic_recognition.py]:
```suggestion
        * the device Pathfinders vtep_ips as destination for non Pathfinders.
```

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py [hunk 2]
```diff
@@ -167,6 +168,11 @@ def _router_bgp_vrfs(self) -> list | None:
                     if (vlan_id := self._mlag_ibgp_peering_vlan_vrf(vrf, tenant)) is not None:
                         self._update_router_bgp_vrf_mlag_neighbor_cfg(bgp_vrf, vrf, tenant, vlan_id)
 
+                # # TODO this is to prevent rendering BGP vrf default on WAN nodes.
+                # # This may need to be adjusted for more complex use cases.
+                # if vrf_name == "default" and self.shared_utils.wan_role is not None:
+                #     continue
+
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py]: Can probably remove this

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/wan_route_servers.schema.yml [hunk 3]
```diff
@@ -25,9 +25,9 @@ keys:
         hostname:
           type: str
           description: Route-Reflector hostname.
-        router_id:
+        vtep_ip:
           type: str
-          description: Route-Reflector router id.
+          description: Route-Reflector VTEP IP Address.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/wan_route_servers.schema.yml]: should we indicate this is the one under DPS interface for clarity for users?

**Review by gmuloc — COMMENTED** [review]
Reviewed all but the route-maps - looks good and make the required changes.
Couple of comments

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py (hunk 2 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/wan_route_servers.schema.yml (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/wan_route_servers.schema.yml]:
```suggestion
          description: Route-Reflector VTEP IP Address. This is usually the IP address under `interface Dps1`.
```

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/wan_route_servers.schema.yml]: @gmuloc is this clearer?

**Review by gmuloc — APPROVED** [review]
LGTM

**Review by carlbuchmann — APPROVED** [review]
LGTM - tested in lab

**Pull request merged**